core/src/main/java/io/questdb/griffin/engine/functions/groupby/AbstractRegressionGroupByFunction.java

```diff
@@ -0,0 +1,148 @@
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.groupby;
+
+import io.questdb.cairo.ArrayColumnTypes;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.map.MapValue;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.engine.functions.DoubleFunction;
+import io.questdb.griffin.engine.functions.GroupByFunction;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.std.Numbers;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * The abstract class, in addition, provides a method to aggregate bivariate/regression statistics.
+ * We use an algorithm similar to B. P. Welford's which works by first aggregating sum of squares of 
+ * independent and dependent variables Sxx = sum[(X - meanX) ^ 2], Syy = sum[(Y - meanY) ^ 2], Sxy = sum[(X - meanX) * (Y - meanY)].
+ * Computation of covariance and correlation is then simple (e.g. covariance = Sxy / (n - 1), correlation = Sxy / sqrt(Sxx * Syy))
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online">Welford's algorithm</a>
+ */
+
+public abstract class AbstractRegressionGroupByFunction extends DoubleFunction implements GroupByFunction, BinaryFunction {
+    protected final Function xFunction;
+    protected final Function yFunction;
+    protected int valueIndex;
+
+    protected AbstractRegressionGroupByFunction(@NotNull Function arg0, @NotNull Function arg1) {
+        this.xFunction = arg0;
+        this.yFunction = arg1;
+    }
+
+    protected void aggregate(MapValue mapValue, double x, double y) {
+        double meanX = mapValue.getDouble(valueIndex);
+        double sumX = mapValue.getDouble(valueIndex + 1);
+        double meanY = mapValue.getDouble(valueIndex + 2);
+        double sumY = mapValue.getDouble(valueIndex + 3);
+        double sumXY = mapValue.getDouble(valueIndex + 4);
+        long count = mapValue.getLong(valueIndex + 5) + 1;
+
+        double oldMeanX = meanX;
+        meanX += (x - meanX) / count;
+        sumX += (x - meanX) * (x - oldMeanX);
```

Comment:
`sumX` and `sumY` are only required in the `corr()` function, so calculating and storing them for the two remaining functions is just a waste of CPU cycles and memory. I'd suggest moving `AbstractRegressionGroupByFunction` code into `AbstractCovarGroupByFunction` (new class) and `CorrGroupByFunction` classes.